### ISSUE TITLE
[Performance]: `stdout.getFile().writer(locking=false)` performs worse than `(new file(1)).writer(locking=false)` in fasta

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#24729]
While working on updating CLBG codes to Chapel 2.0, I came up with what I thought was the clever idea of replacing `(new file(1))` expressions with `stdout.getFile()`, which felt far more comprehensible to the casual reader.  Unfortunately, this change [hurt fasta performance](https://chapel-lang.org/perf/chapcs/?startdate=2024/02/21&enddate=2024/03/29&graphs=submittedfastashootoutbenchmark):

![image](https://github.com/chapel-lang/chapel/assets/7536222/52036c83-5149-476a-b408-bebecb2d3bd2)

I was able to reproduce this on my Mac, just by changing that one expression back and forth.  In the end, I just reverted the code since we'd already been using the similarly obscure `openfd(1)` expression in our last submitted version.

**Comment by bradcray** [comment]
@benharsh looked into this a bit today, but only came up with additional mysteries... (which I'll let him summarize if he thinks they're worth capturing)

**Comment by benharsh** [comment]
For whoever looks into this next (which may be myself):

The issue seems to be the use of ``stdout`` rather than anything involving ``getFile``. Under the hood we open ``stdout`` with the ``FILE*`` from the runtime's ``chpl_cstdout`` function, rather than the integer ``1``. ``chpl_cstdout`` just returns the standard ``stdout`` file descriptor from C. If I use ``(new file(chpl_cstdout()))`` in the test, I also see the poor performance.

It would seem that there's something different about the C-standard ``stdout`` descriptor and ``(new file(1))`` that is causing the worse performance.

**Comment by mppf** [comment]
> It would seem that there's something different about the C-standard `stdout` descriptor and `(new file(1))` that is causing the worse performance.

Yes, that's right. The situation here is that `stdout` is using `FILE*` and line buffering in order to play well with C code that is called (say, as an `extern proc`). There are 2 reasons that is problematic for performance:
 1. Everything is buffered twice. First, we buffer in the Chapel fileWriter / channel; then it is buffered in the `FILE*`, and finally it gets written
 2. We are basically flushing those two buffers every time a newline `\n` is written.

In contrast `new file(1)` does not run into either of the above performance problems.

It would be possible to create a config const / config param / etc to request that `stdout` not include this C interoperability aid. Perhaps it would even be reasonable to toggle it with `--fast`.
